builtin/clone.c

```diff
@@ -929,6 +929,19 @@ static int path_exists(const char *path)
 	return !stat(path, &sb);
 }
 
+/**
+ * Read config variables.
+ */
+static int git_pull_config(const char *var, const char *value, void *cb)
+{
+	if (!strcmp(var, "submodule.recurse")) {
+		option_recurse_submodules.nr = git_config_bool(var, value) ?
+			1 : option_recurse_submodules.nr;

```

Comment:
This is most likely incorrect, as `option_recurse_submodules` is a `string_list`, and increasing the `nr` without even allocating a corresponding item, let alone initializing it, is most likely causing a crash.

Reply:
True, the Size is used as an indication of submodules active or not.

Reply:
But what we really want to do is to stay as close to what passing the `--recurse-submodules` option would have accomplished, which is to add `"."`, not `"true"`...